packages/ember-data/lib/system/relationships/belongs-to.js

```diff
@@ -89,19 +89,21 @@ function belongsTo(modelName, options) {
 
   return computedPolyfill({
     get: function(key) {
-      return this._relationships[key].getRecord();
+      return this.reference._relationships[key].getRecord();
     },
     set: function(key, value) {
       if (value === undefined) {
         value = null;
       }
       if (value && value.then) {
-        this._relationships[key].setRecordPromise(value);
+        this.reference._relationships[key].setRecordPromise(value);
+      } else if (value) {
```

Comment:
No, because either you passed a promise, record or null/undefined. Could add an assert though